src/cf.data.pre

```diff
@@ -7007,14 +7007,12 @@ DOC_END
 
 NAME: esi_parser
 IFDEF: USE_SQUID_ESI
-COMMENT: libxml2|expat|custom
+COMMENT: libxml2|expat
 TYPE: string
 LOC: ESIParser::Type
-DEFAULT: custom
+DEFAULT: auto
 DOC_START
-	ESI markup is not strictly XML compatible. The custom ESI parser
-	will give higher performance, but cannot handle non ASCII character
-	encodings.
+	ESI markup is not strictly XML compatible.
```

Comment:
I'm not sure what that text should be saying now.  My reading of the original was that the custom parser could only handle ASCII strict XML syntax properly, while the other libraries should be better.

Reply:
_If_ the libraries can handle ESI correctly, then the new text could say something like

> Selects the XML parsing library to use when interpreting responses with Edge Side Includes. To disable ESI handling completely, ./configure Squid with --disable-esi.

and

```
DEFAULT_DOC: Selects XXX if available at ./confgure time or YYY otherwise.
```

Please note that I have no idea whether those libraries can handle ESI correctly. My interpretation of the original documentation was that they may not be able to handle some cases, but I did not research that, and other reasonable interpretations (like yours) are possible.

Reply:
looks good. done